### ISSUE TITLE
Go releaser failing with latest update to go 1.19

### ISSUE THREAD

**Issue by ryanchristo** [regen-network/regen-ledger#1698]
## Summary of Bug

https://github.com/regen-network/regen-ledger/actions/runs/3715110559/jobs/6299880767

```
release failed after 12m10s              error=failed to build for darwin_amd64_v1: exit status 2: go: downloading github.com/99designs/go-keychain v0.0.0-20191008050251-8e49817e8af4
```

## Version

`v5.0.0-rc2`

____

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by clevinson** [comment]
Verified that build works locally on both my M1 Mac (which is to be expected), and my old intel Mac (which should be darwin_amd64).  So it looks to be an issue with building for darwin from within a linux environment.

**Comment by clevinson** [comment]
So the issue seems to be independent of whether or not we use cosmos fork of 99designs/keyring (addressed in #1709)

I spent a lot of time digging into this and think it's some problem with how we are doing cross compiling from linux to darwin amd64.  The way we do cross compiling in goreleaser is pulling in a custom fork of osxcross (see [here](https://github.com/regen-network/regen-ledger/blob/main/scripts/release-setup.sh#L11)).

I tried locally updating likhita's fork of osxcross to pull in the latest files from upstream, but that didn't seem to work either.

Today I found this project [goreleaser-cross](https://github.com/goreleaser/goreleaser-cross), which has a prebuilt docker container with osxcross. I think our best bet may be to refactor our goreleaser setup to use this container instead (it seems Akash does the same actually).

If we merge in #1714 for v5.0, then the full resolution of this issue is no longer blocking our upgrade.
